testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/configurationcache/ConfigurationCacheBuildOperationsFixture.groovy

```diff
@@ -35,49 +30,83 @@ class ConfigurationCacheBuildOperationsFixture {
     }
 
     boolean getReused() {
-        storeOperation() == null && loadOperation() != null
+        workGraphStoreOperation() == null && workGraphLoadOperation() != null
     }
 
     void assertStateLoaded() {
-        def load = loadOperation()
-        assertThat(load, notNullValue())
-        assertThat(load.failure, nullValue())
-        assertThat(storeOperation(), nullValue())
+        def load = workGraphLoadOperation()
+        assert load != null && load.failure == null
+        assert workGraphStoreOperation() == null
     }
 
     void assertStateLoadFailed() {
-        def load = loadOperation()
-        assertThat(load, notNullValue())
-        assertThat(load.failure, notNullValue())
-        assertThat(storeOperation(), nullValue())
+        def load = workGraphLoadOperation()
+        assert load != null && load.failure != null
+        assert workGraphStoreOperation() == null
     }
 
     void assertStateStored(boolean expectLoad = true) {
-        def store = storeOperation()
-        assertThat(store, notNullValue())
-        assertThat(store.failure, nullValue())
-        assertThat(loadOperation(), expectLoad ? notNullValue() : nullValue())
+        def store = workGraphStoreOperation()
+        assert store != null && store.failure == null
+        assert (workGraphLoadOperation() != null) == expectLoad
     }
 
     void assertStateStoreFailed() {
-        assertThat(loadOperation(), nullValue())
-        def store = storeOperation()
-        assertThat(store, notNullValue())
-        assertThat(store.failure, notNullValue())
+        def store = workGraphStoreOperation()
+        assert store != null && store.failure != null
+        assert workGraphLoadOperation() == null
+    }
+
+    void assertModelStored() {
+        def modelStore = modelStoreOperation()
+        assert modelStore != null && modelStore.failure == null
+        assert modelLoadOperation() == null
+    }
+
+    void assertModelStoreFailed() {
+        def modelStore = modelStoreOperation()
+        assert modelStore != null && modelStore.failure != null
+        assert modelLoadOperation() == null
+    }
+
+    void assertModelLoaded() {
+        def modelLoad = modelLoadOperation()
+        assert modelLoad != null && modelLoad.failure == null
+        assert modelStoreOperation() == null
     }
 
     void assertNoConfigurationCache() {
-        assertThat(loadOperation(), nullValue())
-        assertThat(storeOperation(), nullValue())
+        assertNoWorkGraphStore()
+        assertNoModelStore()
+    }
+
+    void assertNoWorkGraphStore() {
+        assert workGraphStoreOperation() == null
+        assert workGraphLoadOperation() == null
+    }
+
+    void assertNoModelStore() {
+        assert modelStoreOperation() == null
+        assert modelLoadOperation() == null
+    }
+
+    @Nullable
+    private BuildOperationRecord workGraphStoreOperation() {
+        operations.singleOrNone("Store configuration cache state")
```

Comment:
💭  Ideally, we migrate to use the public build operation type here.

We cannot use a type-based lookup currently because this fixture is used in cross-version tests. The way the underlying operations fixture works is by loading the class from the build operation record, which currently contains only the implementation type. But the implementation types have changed in this PR (`StoreDetails -> WorkGraphStoreDetails`), so this leads to failure in tests.

So, sticking with name-based matching for now.

Related: https://github.com/gradle/gradle/issues/31157